packages/@aws-cdk/aws-ecr-assets/lib/image-asset.ts

```diff
@@ -99,6 +99,10 @@ export class DockerImageAsset extends cdk.Construct implements assets.IAsset {
     this.directory = staging.stagedPath;
     this.sourceHash = staging.sourceHash;
 
+    if (props.target) {
+      this.sourceHash += props.target;
```

Comment:
Makes me think we should also include things like buildArgs and other docker options in the hash, no?

Reply:
probably, though I did notice that the fingerprint function has an "extra" prop to support injecting this info in the hash (vs appending it like i did), but it will require updating StagingProps to pass it through, thoughts?

Reply:
yes, probably the right way to do that (and also add the target to the hash)

Reply:
Any updates here?